easybuild/easyblocks/n/nvhpc.py

```diff
@@ -0,0 +1,201 @@
+        dirs = super(EB_NVHPC, self).make_module_req_guess()
+        for key in dirs:
+            dirs[key] = [os.path.join(self.nvhpc_install_subdir, 'compilers', d) for d in dirs[key]]
+
+        # $CPATH should not be defined in module for NVHPC, it causes problems
+        # cfr. https://github.com/easybuilders/easybuild-easyblocks/issues/830
+        if 'CPATH' in dirs:
+            self.log.info("Removing $CPATH entry: %s", dirs['CPATH'])
+            del dirs['CPATH']
+
+        # EasyBlock Options:
+        ## BYO Compilers allows for using NVHPC's libraries and tools with other, external compilers
+        if self.cfg['module_byo_compilers'] is True:
+            if 'PATH' in dirs:
+                del dirs["PATH"]
+        ## NVHPC is shipped with a compiled OpenMPI installation; enable it by setting the according environment variables
+        if self.cfg['module_nvhpc_own_mpi'] is True:
+            self.nvhpc_mpi_basedir = os.path.join(self.nvhpc_install_subdir, "comm_libs", "mpi")
+            self._nvhpc_extended_components(dirs, self.nvhpc_mpi_basedir, {'PATH': 'bin', 'CPATH': 'include', 'LD_LIBRARY_PATH': 'lib'})
+        ## NVHPC is shipped with math libraries in a dedicated folder; enable them by setting the according environment variables
+        if self.cfg['module_add_math_libs'] is True:
+            self.nvhpc_math_basedir = os.path.join(self.nvhpc_install_subdir, "math_libs")
+            self._nvhpc_extended_components(dirs, self.nvhpc_math_basedir, {'CPATH': 'include', 'LD_LIBRARY_PATH': 'lib64'})
+        ## NVHPC is shipped with NVIDIA's GPU profilers; enable them by setting the according environment variables
+        if self.cfg['module_add_profilers'] is True:
+            self.nvhpc_profilers_basedir = os.path.join(self.nvhpc_install_subdir, "profilers")
+            self._nvhpc_extended_components(dirs, self.nvhpc_profilers_basedir, {'PATH': 'Nsight_Compute'})
+            self._nvhpc_extended_components(dirs, self.nvhpc_profilers_basedir, {'PATH': 'Nsight_Systems/bin'})
+        ## NVHPC is shipped with NCCL; enable it by setting the according environment variables
+        if self.cfg['module_add_nccl'] is True:
+            self.nvhpc_nccl_basedir = os.path.join(self.nvhpc_install_subdir, "comm_libs", "nccl")
+            self._nvhpc_extended_components(dirs, self.nvhpc_nccl_basedir, {'CPATH': 'include', 'LD_LIBRARY_PATH': 'lib'})
+        ## NVHPC is shipped with NVSHMEM; enable it by setting the according environment variables
+        if self.cfg['module_add_nvshmem'] is True:
+            self.nvhpc_nvshmem_basedir = os.path.join(self.nvhpc_install_subdir, "comm_libs", "nvshmem")
+            self._nvhpc_extended_components(dirs, self.nvhpc_nvshmem_basedir, {'CPATH': 'include', 'LD_LIBRARY_PATH': 'lib'})
+        ## NVHPC is shipped with CUDA; rather use this CUDA than an external CUDA module (via $CUDA_HOME) by setting the according environment variables
```

Comment:
Re 2#: I'm using Markdown-styled two levels of headers. There are two hashes because I consider these comments a level below `# EasyBlock Options:`.
Yay/nay?

Reply:
We consistently use a single hash for comments, otherwise our code style checks will keep failing...